OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -56,19 +56,25 @@ static string[] QueryDevices(string label, int type)
 
 			var devices = new List<string>();
 			var next = ALC10.alcGetString(IntPtr.Zero, type);
+			if (next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)
+			{
+				Log.Write("sound", "Failed to query OpenAL device list using {0}", label);
+				return new string[] { };
+			}
+
 			do
 			{
 				var str = Marshal.PtrToStringAnsi(next);
+				if (string.IsNullOrEmpty(str))
```

Comment:
I also know about that, which is why my first attempt was to just catch there, but apparently it isn't enough. What changes do you request in particular?